src/System.Management.Automation/engine/Utils.cs

```diff
@@ -2312,6 +2312,14 @@ internal static void NotNullOrEmpty(string value, string paramName)
             }
         }
 
+        internal static void NotNullOrEmpty(ICollection value, string paramName)
```

Comment:
Perhaps use `[CallerArgumentExpression("value")]` ?

Reply:
It's not implemented yet, right? See the [proposal doc](https://github.com/dotnet/csharplang/blob/master/proposals/caller-argument-expression.md) and the open issue: https://github.com/dotnet/csharplang/issues/287

Reply:
You're correct! I was misled by the docs.